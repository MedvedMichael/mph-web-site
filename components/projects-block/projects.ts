import { Project } from "./project-card";

export const projects: Project[] = [
    {
        text:"Full-stack project with full MERN stack of technologies!\nIt can be customized for storing lots of videos, photos; its structure is suitable for the majority of situations. Moreover, after the completion of the project I successfully deployed and configured all of its parts to remote server!!!ðŸ˜ŽðŸ˜ŽðŸ˜Ž",
        image:'/pictures/web-storage.png',
        title:'Web Storage',
        src: ['https://github.com/MedvedMichael/web-storage-client', 'https://github.com/MedvedMichael/web-storage']
    },
    {
        text:"Telegram Bot that is created for people who want to learn English or other languages!\nIt's based on the famous method of using flash cards - cards bearing information on both sides, which is intended to be used as an aid in memorization",
        image:'/pictures/flash-cards.png',
        title:'Flash Cards Bot',
        src: ['https://github.com/MedvedMichael/flash-cards-bot']
    },
    {
        text:"Yeah, this site is completely created by me!)))\nI used Next.js as a framework for frontend using SSR (Server-Side Rendering) and Firebase from Google to store my data.\nAlso I didn't use any CSS-libraries like Bootstrap on MaterialUI, and I implemented the dark-mode feature using styled components (that was my dreamðŸ˜‰)!!!",
        image:'/pictures/nextjs.jpg',
        title:'My web-site',
        src: ['https://github.com/MedvedMichael/mph-web-site']
    },
    // {
    //     text:"Full-stack project with full MERN stack of technologies! It can be customized for storing lots of videos, photos; its structure is suitable for the majority of situations",
    //     image:'/pictures/web-storage.png',
    //     title:'Web Storage'
    // }
]