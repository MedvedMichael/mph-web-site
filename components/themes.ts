const light = {
    bg: {
        primary: '#c8d8e3',
        secondary: '#ffffff',
        inset: '#e2e4e8',
        input: 'rgba(65,67,78,0.12)',
        nav: '#5785ba'
    },
    text: {
        primary: '#fbfbfc',
        secondary: '#2f3037',
        tertiary: '#525560',
        quarternary: '#9194a1',
        placeholder: 'rgba(82,85,96,0.5)',
        onPrimary: '#ffffff'
    },
    border: {
        nav: '1px solid #282828'
    }
    // ...
}

const dark = {
    bg: {
        primary: '#162333',
        secondary: '#111111',
        inset: '#111111',
        input: 'rgba(191,193,201,0.12)',
        nav: '#18304d'
    },
    text: {
        primary: '#fbfbfc',
        secondary: '#e3e4e8',
        tertiary: '#a9abb6',
        quarternary: '#6c6f7e',
        placeholder: 'rgba(145,148,161,0.5)',
        onPrimary: '#050505'
    },
    border: {
        nav: '1px solid #282828'
    }
    // ...
}

const defaultTheme = {
    fontSizes: [
        '14px', // 0
        '16px', // 1
        '18px', // 2
        '22px', // 3
        '26px', // 4
        '32px', // 5
        '40px'  // 6
    ],
    fontWeights: {
        body: 400,
        subheading: 500,
        link: 600,
        bold: 700,
        heading: 800,
    },
    lineHeights: {
        body: 1.5,
        heading: 1.3,
        code: 1.6,
    },
    // ...
};

export const lightTheme = { ...defaultTheme, ...light }
export const darkTheme = { ...defaultTheme, ...dark }