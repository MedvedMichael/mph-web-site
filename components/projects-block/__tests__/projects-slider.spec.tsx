import { shallow } from 'enzyme'
import ProjectsSlider, { DotView } from '../projects-slider'
import {projects} from '../projects'
import * as React from 'react';
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { lightTheme } from '../../themes'
import 'jest-styled-components'
import toJson from 'enzyme-to-json';

test('should render', () => {
    const res = shallow(<ProjectsSlider projects={projects}/>)
    expect(toJson(res)).toMatchSnapshot()
})

const mockSetState = jest.fn()

jest.mock('react', () => ({
    ...jest.requireActual('react'),
    useState: initial => [initial, mockSetState]
}));

test ('should set selected dot on click', () => {
    const res = shallow(<ProjectsSlider projects={projects}/>)
    res.find(DotView).first().simulate('click')
    expect(mockSetState).toBeCalledTimes(1)
})

test ('should have correct styles', () => {
    const res = renderer.create(<ThemeProvider theme={lightTheme}><DotView/></ThemeProvider>).toJSON()
    expect(res).toHaveStyleRule('transition', lightTheme.transition.bg)
})