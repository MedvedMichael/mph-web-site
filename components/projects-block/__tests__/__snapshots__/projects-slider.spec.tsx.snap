// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div>
  <styled.div>
    <ProjectCard
      image="/pictures/web-storage.png"
      key="card0"
      selected={true}
      src={
        Array [
          "https://github.com/MedvedMichael/web-storage-client",
          "https://github.com/MedvedMichael/web-storage",
        ]
      }
      text="Full-stack project with full MERN stack of technologies!
It can be customized for storing lots of videos, photos; its structure is suitable for the majority of situations. Moreover, after the completion of the project I successfully deployed and configured all of its parts to remote server!!!ðŸ˜ŽðŸ˜ŽðŸ˜Ž"
      title="Web Storage"
    />
  </styled.div>
  <styled.div>
    <styled.div
      key="dot0"
      onClick={[Function]}
      style={
        Object {
          "background": "#3160d6",
        }
      }
    />
    <styled.div
      key="dot1"
      onClick={[Function]}
      style={Object {}}
    />
    <styled.div
      key="dot2"
      onClick={[Function]}
      style={Object {}}
    />
  </styled.div>
</div>
`;
